test/import-json.test.ts

```diff
@@ -688,28 +937,245 @@ describe('importJson', () => {
 
 		expect(importEvents.trigger).toHaveBeenNthCalledWith(
 			1,
-			'percentage-update',
+			'percentage-import',
 			20
 		);
 		expect(importEvents.trigger).toHaveBeenNthCalledWith(
 			2,
-			'percentage-update',
+			'percentage-import',
 			40
 		);
 		expect(importEvents.trigger).toHaveBeenNthCalledWith(
 			3,
-			'percentage-update',
+			'percentage-import',
 			60
 		);
 		expect(importEvents.trigger).toHaveBeenNthCalledWith(
 			4,
-			'percentage-update',
+			'percentage-import',
 			80
 		);
 		expect(importEvents.trigger).toHaveBeenNthCalledWith(
 			5,
-			'percentage-update',
+			'percentage-import',
 			100
 		);
 	});
+
+	test('should update UUID map when enableInternalLinks is true', async () => {
+		// Clear mocks before running this test
+		jest.clearAllMocks();
+
+		// Reset the mock implementations
+		mockUuidMapStore.read.mockReset();
+		mockUuidMapStore.write.mockReset();
+
+		vault.getAbstractFileByPath.mockImplementation((path: string) => {
+			if (path === 'day-one-in') {
+				return { children: [fakeJsonFile] } as unknown as TFolder;
+			}
+			if (path === 'day-one-in/fakeFile.json') {
+				return fakeJsonFile;
+			}
+			return null;
+		});
+
+		// Simulate the vault.create behavior
+		vault.create.mockImplementation(() => Promise.resolve({} as TFile));
+		vault.read.mockResolvedValue(
+			JSON.stringify({
+				entries: [
+					{
+						...mockEntry,
+						uuid: 'abc123',
+					},
+					{
+						...mockEntry,
+						uuid: 'def456',
+					},
+				],
+			})
+		);
+
+		// Simulate an existing UUID map
+		const existingUuidMap = {
+			'1F9422B8806C435E8AA24FB5DAE47F5D': 'old-file.md',
+		};
+		mockUuidMapStore.read.mockResolvedValue(existingUuidMap);
+
+		await importJson(
+			vault,
+			{
+				...DEFAULT_SETTINGS,
+				inDirectory: 'day-one-in',
+				inFileName: 'fakeFile.json',
+				enableInternalLinks: true, // Enable internal links
+			},
+			fileManager,
+			importEvents,
+			mockUuidMapStore
+		);
+
+		// Verify that read was called
+		expect(mockUuidMapStore.read).toHaveBeenCalled();
+
+		// Verify that write was called
+		expect(mockUuidMapStore.write).toHaveBeenCalled();
+
+		// Verify the map was updated with our entries
+		// Get the first argument of the first call to write
+		const updatedMap = mockUuidMapStore.write.mock.calls[0][0];
+
+		// Verify that the existing entry was preserved
+		expect(updatedMap['1F9422B8806C435E8AA24FB5DAE47F5D']).toBe('old-file.md');
+
+		// Verify that both new UUIDs were added to the map
+		expect(updatedMap).toHaveProperty('abc123');
+		expect(updatedMap).toHaveProperty('def456');
+	});
+
+	test('should NOT update UUID map when enableInternalLinks is false', async () => {
+		// Clear mocks before running this test
+		jest.clearAllMocks();
+
+		// Reset the mock implementations
+		mockUuidMapStore.read.mockReset();
+		mockUuidMapStore.write.mockReset();
```

Comment:
These resets should be moved into the `afterEach` so that it's guaranteed nothing leaks between tests